tests/loggers/test_cometml_logger.py

```diff
@@ -76,16 +77,42 @@ def test_comet_ml_log_image_saves_images(images: torch.Tensor, channels_last: bo
     assert actual_num_images == expected_num_images
 
 
-@pytest.mark.parametrize(
-    'images,channels_last',
-    [
-        (torch.rand(32, 0), False),  # Has zero in dimension.
-        (torch.rand(4, 4, 8, 32, 32), False),  # > 4 dim.
-        ([torch.rand(4, 32, 32, 3)], True),
-    ])  # sequence > 3 dim.
-def test_comet_ml_log_image_errors_out(comet_logger, images, channels_last):
-    with pytest.raises(ValueError):
-        comet_logger.log_images(images, channels_last=channels_last)
+@pytest.mark.parametrize('images,masks', [(torch.randint(0, 256, (32, 32, 3)), {
+    'pred': torch.randint(0, 10, (32, 32))
+}), (torch.rand(4, 32, 32, 3), {
+    'pred': torch.randint(0, 10, (4, 32, 32))
+}), (torch.rand(4, 32, 32, 3), {
+    'pred': 4 * [torch.randint(0, 10, (32, 32))]
+}), (torch.rand(4, 32, 32, 3), {
+    'pred': torch.randint(0, 10, (4, 32, 32)),
+    'pred2': torch.randint(0, 10, (4, 32, 32))
+})])
+def test_comet_ml_log_image_saves_images_with_masks(images: torch.Tensor, masks: Dict[str, torch.Tensor],
+                                                    comet_logger: CometMLLogger, comet_offline_directory: str):
+    assert isinstance(comet_offline_directory, str)
+    # Count expected images and generate numpy arrays from torch tensors.
+    num_masks = len(masks.keys())
+    num_images = images.shape[0] if images.ndim == 4 else 1
+    expected_num_images = num_images * (2 * num_masks + 1)
```

Comment:
Why is this the expected number of images?

Reply:
For each image, we plot the image (num_images * 1) + the image with the mask overlaid and the mask by itself for each mask: (num_images * 2 * num_masks).

Reply:
done